pkgs/development/compilers/rust/cargo_cross.nix

```diff
@@ -0,0 +1,23 @@
+
+{stdenv, pkgsBuildBuild, makeShellWrapper, cacert, rustc}:
+
+stdenv.mkDerivation {
+  name = "${stdenv.targetPlatform.config}-cargo";
+  version = "0";
+
+  # Use depsBuildBuild or it tries to use target-runtimeShell
+  depsBuildBuild = [ makeShellWrapper ];
+
+  # Avoid missing src error
+  unpackPhase = "true";
+
+  installPhase = ''
+    mkdir -p $out/bin
+    ln -vs ${pkgsBuildBuild.cargo}/bin/.cargo-wrapped $out/bin/cargo
+
+    wrapProgram "$out/bin/cargo" \
+      --suffix PATH : "${rustc}/bin" \
+      --set CARGO_HTTP_CAINFO "${cacert}/etc/ssl/certs/ca-bundle.crt" \
+      --set SSL_CERT_FILE "${cacert}/etc/ssl/certs/ca-bundle.crt"
```

Comment:
Maybe it would be better to just re-wrap `${pkgsBuildBuild.cargo}/bin/cargo`?  That way you wouldn't need to specify the SSL, which I'm worried would otherwise go out of sync.